conditional-process3.nf

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env nextflow
+
+/*
+ * Copyright (c) 2023, Seqera Labs.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+ /*
+  * author Ben Sherman <bentshermann@gmail.com> 
+  */
+
+process foo {
+    input:
+    val signal
+    output: 
+    path 'x.txt'
+
+    script:
+    '''
+    echo foo > x.txt
+    '''
+}
+
+process bar {
+    input:
+    val signal
+    output: 
+    path 'x.txt'
+
+    script:
+    '''
+    echo bar > x.txt
+    '''
+}
+
+workflow {
+    ch_if = Channel.of( 1..100 )
+      | randomSample(1)
+      | branch { n ->
+        TRUE: n > 50
+        FALSE: n <= 50
+      }
```

Comment:
I'd only suggest to not using `TRUE` and `FALSE` which can be easily confused with `true` and `false`. Actually in the description, it was mentioned the latter. May `GREEN` and `BLUE` just to stress are not lang keywords?

Reply:
That's why I made it uppercase, so that it doesn't conflict with the keywords. I don't think colors would work because different cultures use different color conventions. Like in the US we would probably use green and red.

Reply:
Using colors was just a suggestion that was not true/false. Whatever you prefer. `foo` and `bar`, or `alpha` and `beta`, whatever

Reply:
Okay I will change to `high` and `low`